File: backend/SNS.Application/Jobs/Companies/Contracts/CompanyDetailsDto.cs
using SNS.Domain.Jobs.Enums;

namespace SNS.Application.Jobs.Companies.Contracts;

public sealed record CompanyDetailsDto(
    Guid Id,
    string Name,
    string Industry,
    string? WebsiteUrl,
    string? LogoUrl,
    DateTime CreatedAt,
    bool IsActive,
    int ActiveJobsCount,
    int AdministratorsCount,
    CompanyRole? CurrentUserRole
);

File: SNS.Web/src/app/jobs/companies/contracts/company-details.dto.ts
import { CompanyRole } from '../../enums/company-role.enum';

export interface CompanyDetailsDto {
    id: string;
    name: string;
    industry: string;
    websiteUrl: string | null;
    logoUrl: string | null;
    createdAt: string;
    updatedAt: string;
    isActive: boolean;
    administratorsCount: number;
    activeJobsCount: number;
    currentUserRole?: CompanyRole | null;
}
